### PR TITLE
Prevent pending registrations from being checked-in

### PR DESCRIPTION
**Pull request by OmeGak**

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -18,6 +18,8 @@ Bugfixes
 
 - Only show the warning about draft mode in a conference if it actually has any
   contributions or timetable entries
+- Prevent pending registrations to be marked as checked-in (:pr:`4646`, thanks
```
Comment (ThiefMaster) [inline CHANGES.rst:21]: s/to be/from being/?

File: indico/modules/events/registration/controllers/management/reglists.py [hunk 2]
```diff
@@ -602,6 +603,8 @@ class RHRegistrationCheckIn(RHManageRegistrationBase):
     """Set checked in state of a registration"""
 
     def _process_PUT(self):
+        if self.registration.state not in (RegistrationState.complete, RegistrationState.unpaid):
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/management/reglists.py:606]: Is this the same logic we use to hide the UI for it?

Comment (OmeGak) [reply indico/modules/events/registration/controllers/management/reglists.py:606]: [Yep].(https://github.com/indico/indico/blob/master/indico/modules/events/registration/templates/management/_registration_details.html#L20)

**Review by ThiefMaster — APPROVED** [review]

File: indico/modules/events/registration/controllers/management/reglists.py [hunk 3]
```diff
@@ -602,6 +603,8 @@ class RHRegistrationCheckIn(RHManageRegistrationBase):
...
+            raise BadRequest(_('Pending registration cannot be marked as checked-in'))
```
Comment (ThiefMaster) [inline indico/modules/events/registration/controllers/management/reglists.py:607]: s/Pending/This/ since it's not necessarily pending?

**Review by ThiefMaster — APPROVED** [review]

**Pull request merged**